lib/units/validateBody.js

```diff
@@ -148,6 +148,27 @@ function getBodyValidator(realType, expectedType) {
   return [null, validator[0]];
 }
 
+/**
+ * Parses a given JSON Schema to ensure it's a valid JSON.
+ * @param {string} schema
+ */
+const validateJsonSchema = (schema) => {
+  try {
+    JSON.parse(schema);
+  } catch (error) {
+    throw new Error(`\
+Failed to parse a given JSON Schema:
+
+${schema}
+
+Please check if that is a valid JSON and a valid JSON Schema.
+
+Original parsing error:
+${error.stack}\
```

Comment:
If I understand correctly, this message would make it to the user, perhaps it isn't a good idea to show them our internal stack trace.

Reply:
I didn't realize this would include our stack trace as well. Do you think removing stack trace for good would be a viable option here?

Reply:
I think `error.stack` will include the full stack, perhaps in the case of the traffic inspector or a hosted closed source product, the stack trace could contain implementation details outside gavel.

I am not sure the stack will give the user much help in fixing their problem, but `error.message` probably will, it should contain the underlying cause and source map information.

For example:

```
> let error;
> try { JSON.parse('{"href"}') } catch (e) { error = e }
SyntaxError: Unexpected token } in JSON at position 7
    at JSON.parse (<anonymous>)
    at repl:1:12
    at Script.runInThisContext (vm.js:96:20)
    at REPLServer.defaultEval (repl.js:332:29)
    at bound (domain.js:395:14)
    at REPLServer.runBound [as eval] (domain.js:408:12)
    at REPLServer.onLine (repl.js:639:10)
    at REPLServer.emit (events.js:194:15)
    at REPLServer.EventEmitter.emit (domain.js:441:20)
    at REPLServer.Interface._onLine (readline.js:290:10)

> error
SyntaxError: Unexpected token } in JSON at position 7
    at JSON.parse (<anonymous>)
    at repl:1:12
    at Script.runInThisContext (vm.js:96:20)
    at REPLServer.defaultEval (repl.js:332:29)
    at bound (domain.js:395:14)
    at REPLServer.runBound [as eval] (domain.js:408:12)
    at REPLServer.onLine (repl.js:639:10)
    at REPLServer.emit (events.js:194:15)
    at REPLServer.EventEmitter.emit (domain.js:441:20)
    at REPLServer.Interface._onLine (readline.js:290:10)

> error.message
'Unexpected token } in JSON at position 7'

> error.stack
'SyntaxError: Unexpected token } in JSON at position 7\n    at JSON.parse (<anonymous>)\n    at repl:1:12\n    at Script.runInThisContext (vm.js:96:20)\n    at REPLServer.defaultEval (repl.js:332:29)\n    at bound (domain.js:395:14)\n    at REPLServer.runBound [as eval] (domain.js:408:12)\n    at REPLServer.onLine (repl.js:639:10)\n    at REPLServer.emit (events.js:194:15)\n    at REPLServer.EventEmitter.emit (domain.js:441:20)\n    at REPLServer.Interface._onLine (readline.js:290:10)'
```

Reply:
I've originally included `error.message` only, but thought it would have been useful for the end users to see _their_ stack trace to the exception. I would leave error message only for now.

Reply:
Resolved in 7342a7a.